HelpSource/Reference/Syntax-Shortcuts.schelp

```diff
@@ -11,38 +11,55 @@ Because of the multiple syntax equivalences, some expressions can be written in
 code::
 // new argument syntax
 
-(1..10).collect({|n| n.squared }); // receiver syntax
+(1..10).collect({|n| n.squared });  // receiver syntax
 collect((1..10), {|n| n.squared }); // function call syntax
-(1..10).collect {|n| n.squared }; // receiver syntax with trailing function arg
-collect ((1..10)) {|n| n.squared }; // function call syntax with trailing function arg
-(1..10) collect: {|n| n.squared }; // binary operator syntax
+(1..10).collect {|n| n.squared };   // receiver syntax with trailing block argument
+collect ((1..10)) {|n| n.squared }; // function call syntax with trailing block argument
+(1..10) collect: {|n| n.squared };  // binary operator syntax
 
 
 // old argument syntax
 
-(1..10).collect({ arg n; n.squared }); // receiver syntax
+(1..10).collect({ arg n; n.squared });  // receiver syntax
 collect((1..10), { arg n; n.squared }); // function call syntax
-(1..10).collect { arg n; n.squared }; // receiver syntax with trailing function arg
-collect ((1..10)) { arg n; n.squared }; // function call syntax with trailing function arg
-(1..10) collect: { arg n; n.squared }; // binary operator syntax
+(1..10).collect { arg n; n.squared };   // receiver syntax with trailing block arg
+collect ((1..10)) { arg n; n.squared }; // function call syntax with trailing block arg
+(1..10) collect: { arg n; n.squared };  // binary operator syntax
 
 
 // partial application syntax
 
 (1..10).collect( _.squared ); // receiver syntax
 collect((1..10), _.squared ); // function call syntax
-(1..10) collect: _.squared ; // binary operator syntax
+(1..10) collect: _.squared ;  // binary operator syntax
 ::
 
 You could even start expanding out the equivalent of (1..10) which is really a shortcut for code:: series(1, nil, 10) ::. This could also be written code:: 1.series(nil,10) ::. This adds another 26 variations to the 13 variations above.
 
-section:: Objects, functions, messages and arguments
+section:: Defining functions and classes
 
-subsection:: functional and receiver notation
+subsection:: shorter argument lists in definitions
 table::
 ## instead of writing: || you can write:
-## code:: f(x, y) :: || code:: x.f(y) ::
-## code:: f(g(x)) :: || code:: x.g.f ::
+## code:: { arg x; x < 2 } :: || code:: {|x| x < 2 } ::
+## code:: { arg x = 123; x < 2 } :: || code:: {|x = 123| x < 2 } ::
+## code:: { arg x = 10.rand; x < 2 } :: || code:: {|x = (10.rand)| x < 2 } :: or code:: {|x(10.rand)| x < 2 } ::
+::
+note::
+When using the new code::||:: syntax, the default value needs to be enclosed in parenthesis if it's not a literal.
+::
+
+subsection:: partial application
+table::
+## instead of writing: || you can write:
+## code:: {|x| object.msg(a, x, b) } :: || code:: object.msg(a, _, b) ::
+## code:: {|x,y| object.msg(a, x, y) } :: || code:: object.msg(a, _, _) ::
+## code:: {|x| a + x } :: || code:: a + _ ::
+## code:: {|x| [a, b, x] } :: || code:: [a, b, _] ::
+## code:: {|x| (a: x) } :: || code:: (a: _) ::
+::
+warning::
```

Comment:
this is probably just OK as a note. i'd like to reserve warnings for things that are really dangerous, and use notes for things that are surprising or unintuitive.

Reply:
OK.